### PR TITLE
Issue166 redcap form status

### PR DESCRIPTION
**Pull request by szh2425**
Add completion field to redcap forms and display and cache completion status on record table.

- in drivers/redcap/formbuilder.py, create method add_completion_field (acc740d)
- in drivers/redcap/driver.py, create method find_completed_forms (5d005818d351956e73f84e5c637da81ade84f821, b73347b5123aa84cfdffc11870f7ce48486b5ab4 08e4cfe15f241efc77327c7d959f5fde01846a66, acc740dcc3b978d2b439ede9300001c8470d502b, affab02e2b3c112ed04a3426d7f4463e0e89dbd5)
- in drivers/redcap/driver.py, add colors to "edit" buttons (0aad9ce81e64304cb3c183b9c848ab467da8cba2, 409db920c3abd21e3cf9c4ce1249ec78e0f2525f)
- in tests/unit_tests/test_form_builder.py, modify test_construct_form to assert field was added (5cfc923b64090762576eff25b3d91ecd69cdc6b8)
- in tests/unit_tests/test_redcap_driver.py, add 3 unit tests for find_completion_codes (5155a0d7827d319ab4d1a36243cba818a2af4089)

### CODE REVIEW

**Comment by r3m0chop** [comment]
Having just seen this for the first time, I am curious if this well needed "Color coding redcap record table" has taken into account any accessibility issues surrounding those with color blindness that might not allow for the user to distinguish such variations in color?  

For your consideration, rather than merely setting `style="background-color:<color>"` (or its updated `class="btn btnsmall btn-<success_status>" `), you may want to also consider including an additional icon, or (likely for most accessibility) at least providing alternate wording such as `Update` rather than `Edit` where applicable.  

I might further recommend that you  `.format()` these all together into a single `string_to_return` variable with each if, rather than doing the format at each return, in order to improve maintainability.   And, along the lines of maintainability, I also note that this same set of code seems to appear in a couple different places in `ehb_datasources/drivers/redcap/driver.py` that could be consolidated and possibly even incorporate more self-documenting variable names than `i`, `j`, and `l`, for example.  But I digress, and will aim to save such code-directed details for the actual code review.  Just wanted to emphasize the very "accessible" possibilities of including further accessibility into the representation.

**Comment by szh2425** [comment]
@r3m0chop Great suggestions - thank you! I will try to address some of them before the code review so we don't have to go over it twice. As for color blindness, interesting and definitely applicable point. I will look into different alternatives/additions to the edit buttons.

File: ehb_datasources/drivers/redcap/driver.py [hunk 1]
```diff
@@ -629,70 +631,90 @@ def counter(start):
...
+            def makeRow(form_name, form_index):
+                key = str(form_index)
                 row = '<tr><td>' + reduce(
                     lambda x,
                     y: x + ' ' + y.capitalize(),
-                    fn.split('_'), '') + '</td>'
+                    form_name.split('_'), '') + '</td>'
                 return row + ('<td><button data-toggle="modal"' +
-                              ' data-backdrop="static" data-keyboard="false"' +
-                              ' href="#pleaseWaitModal" class="btn btn-small' +
```
Comment (seg1129) [inline ehb_datasources/drivers/redcap/driver.py]: is it possible to update this to use the python .join function? more information can be found [here](https://www.geeksforgeeks.org/join-function-python/)

Comment (szh2425) [reply ehb_datasources/drivers/redcap/driver.py]: yes - f3a23bc6e93fab826539b23658346b5cbd94ffb1

**Pull request merged**